lib/gis/file_name.c

```diff
@@ -44,9 +65,30 @@ char *G_file_name(char *path,
 /*!
   \brief Builds full path names to GIS misc data files
 
-  \param[out] path buffer to hold resultant full path to file
-  \param dir misc directory
-  \param element database element (eg, "cell", "cellhd", "vector", etc)
+  Paths to misc files are in form:
+  /home/user/GISDBASE/LOCATION/mapset/dir/name/element
+
+  path input buffer memory must be preallocated by caller.
+
+  C:
+  @code
+  char path[GPATH_MAX];
+  @endcode
+  Python:
+  @code
+  import ctypes
+  from grass.pygrass.utils import decode
+  from grass.lib.gis import G_file_name_misc, GPATH_MAX
+
+  path = ctypes.create_string_buffer(GPATH_MAX)
+  path_str = decode(G_file_name_misc(path, "dir", "elem", "name", "mapset"))
+  print(path_str)
+  /home/user/gisdb/loc/mapset/dir/name/elem
+  @endcode
+
+  \param[out] path preallocated buffer to hold resultant full path to file
+  \param dir misc directory (e.g., "cell_misc", "group")
+  \param element database element (e.g., "history", "REF")
   \param name name of file to build path to (fully qualified names allowed)
```

Comment:
Isn't _file_ here the confusing part? Name is name of raster map (or whatever we call it), not of a file. I think we lack a good term for that named thing, so that's why I went with _object_ in #1681, but I don't consider that naming issue resolved.

Reply:
You are right. I should have read twice. I went with object until we come up with a new naming scheme.